File: Data Structures & Algorithms/combination-target-sum/submission-1.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res, path = [], []
        n = len(nums)
        def backtesting(startindex, total):
            nonlocal res, path, target, n
            if total > target: 
                return
            if total == target:
                res.append(path[:])
                return
            for i in range(startindex,n):
                path.append(nums[i])
                total += nums[i]
                backtesting(i,total)
                path.pop()
                total -= nums[i]
        backtesting(0, 0)
        return res

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-1.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int i = 0, j = 0;
        int n = nums.size();
        int target = 0;
        while (j<n){
            target = nums[j];
            nums[i] = target;
            //++j;
            ++i;
            
            while (j<n && nums[j]==target){
                ++j;
            }

        }
        return i;

    }
};